sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SdkSpan.java

```diff
@@ -449,8 +460,57 @@ public ReadWriteSpan recordException(Throwable exception, Attributes additionalA
       additionalAttributes = Attributes.empty();
     }
 
-    addTimedEvent(
-        ExceptionEventData.create(spanLimits, clock.now(), exception, additionalAttributes));
+    AttributesBuilder attributesBuilder = Attributes.builder();
+    String exceptionName = exception.getClass().getCanonicalName();
+    String exceptionMessage = exception.getMessage();
+    StringWriter stringWriter = new StringWriter();
+    try (PrintWriter printWriter = new PrintWriter(stringWriter)) {
+      exception.printStackTrace(printWriter);
+    }
+    String stackTrace = stringWriter.toString();
+
+    if (this.spanLimits.getMaxNumberOfAttributes() > 0
+        && this.spanLimits.getMaxAttributeValueLength() != Integer.MAX_VALUE) {
+      if (exceptionName != null) {
+        exceptionName =
+            exceptionName.substring(
+                0, Math.min(exceptionName.length(), this.spanLimits.getMaxAttributeValueLength()));
+      }
+
+      if (exceptionMessage != null) {
+        exceptionMessage =
+            exceptionMessage.substring(
+                0,
+                Math.min(exceptionMessage.length(), this.spanLimits.getMaxAttributeValueLength()));
+      }
+
+      if (stackTrace != null) {
+        stackTrace =
+            stackTrace.substring(
+                0, Math.min(stackTrace.length(), this.spanLimits.getMaxAttributeValueLength()));
+      }
+    }
+
+    if (exceptionName != null) {
+      attributesBuilder.put(EXCEPTION_TYPE, exceptionName);
+    }
+    if (exceptionMessage != null) {
+      attributesBuilder.put(EXCEPTION_MESSAGE, exceptionMessage);
+    }
+    if (stackTrace != null) {
+      attributesBuilder.put(EXCEPTION_STACKTRACE, stackTrace);
+    }
+
+    attributesBuilder.putAll(additionalAttributes);
+
+    AttributeUtil.applyAttributesLimit(
+        attributesBuilder.build(),
+        spanLimits.getMaxNumberOfAttributesPerEvent(),
+        spanLimits.getMaxAttributeValueLength());
+
+    Attributes attributes = attributesBuilder.build();
+
+    addTimedEvent(ExceptionEventData.create(clock.now(), exception, attributes));
```

Comment:
We have a class called `AttributesMap` which applies span limits as entries are added:
```suggestion
    AttributesMap attributes =AttributesMap.create(spanLimits.getMaxNumberOfAttributes(), spanLimits.getMaxAttributeValueLength());
    String exceptionName = exception.getClass().getCanonicalName();
    String exceptionMessage = exception.getMessage();
    StringWriter stringWriter = new StringWriter();
    try (PrintWriter printWriter = new PrintWriter(stringWriter)) {
      exception.printStackTrace(printWriter);
    }
    String stackTrace = stringWriter.toString();

    if (exceptionName != null) {
      attributes.put(EXCEPTION_TYPE, exceptionName);
    }
    if (exceptionMessage != null) {
      attributes.put(EXCEPTION_MESSAGE, exceptionMessage);
    }
    if (stackTrace != null) {
      attributes.put(EXCEPTION_STACKTRACE, stackTrace);
    }

    additionalAttributes.forEach(attributes::put);

    addTimedEvent(ExceptionEventData.create(clock.now(), exception, attributes));
```

Reply:
Cool, I didn't know this, this is cool.